src/stores/sounds.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * SPDX-FileCopyrightText: 2019 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { defineStore } from 'pinia'
+import Vue from 'vue'
+
+import { getCurrentUser } from '@nextcloud/auth'
+import { loadState } from '@nextcloud/initial-state'
+import { generateFilePath } from '@nextcloud/router'
+
+import BrowserStorage from '../services/BrowserStorage.js'
+import { setPlaySounds } from '../services/settingsService.js'
+
+/**
+ * Get play sounds option (from server for user or from browser storage for guest)
+ * @param {boolean} value whether sounds should be played
+ */
+const hasUserAccount = Boolean(getCurrentUser()?.uid)
+const shouldPlaySounds = hasUserAccount
+	? loadState('spreed', 'play_sounds', false)
+	: BrowserStorage.getItem('play_sounds') !== 'no'
+
+/**
+ * Preferred version is the .ogg, with .flac fallback if .ogg is not supported (Safari)
+ */
+const fileExtension = new Audio().canPlayType('audio/ogg') ? '.ogg' : '.flac'
+
+export const useSoundsStore = defineStore('sounds', {
+	state: () => ({
+		shouldPlaySounds,
+		audioObjectsCreated: false,
+		audioObjects: {
+			join: null,
+			leave: null,
+			wait: null,
+		},
+	}),
+
+	actions: {
+		/**
+		 * Set play sounds option (on server for user or in browser storage for guest)
+		 * @param {boolean} value whether sounds should be played
+		 */
+		async setShouldPlaySounds(value) {
+			await setPlaySounds(hasUserAccount, value ? 'yes' : 'no')
+			this.shouldPlaySounds = value
+		},
+
+		playAudio(key) {
+			if (!this.audioObjectsCreated) {
+				this.initAudioObjects()
+			}
+			this.audioObjects[key].play()
+		},
+
+		pauseAudio(key) {
+			if (!this.audioObjectsCreated) {
+				this.initAudioObjects()
+			}
```

Comment:
```suggestion
			if (!this.audioObjectsCreated) {
				return
			}
```

If it wasn't initiated before, it means it wasn't needed and maybe it's not needed after pausing

Reply:
I'd keep the first one. We are initiating objects only if CallButton is clicked, so it will act as fallback, if for some reason it wouldn't be registered